packages/jest-haste-map/src/haste_fs.js

```diff
@@ -15,45 +15,49 @@ import micromatch from 'micromatch';
 import H from './constants';
 
 export default class HasteFS {
+  _rootDir: Path;
   _files: FileData;
 
-  constructor(files: FileData) {
+  constructor({rootDir, files}: {rootDir: Path, files: FileData}) {
+    this._rootDir = rootDir;
     this._files = files;
   }
 
   getModuleName(file: Path): ?string {
-    const fileMetadata = this._files.get(file);
+    const fileMetadata = this._getFileData(file);
     return (fileMetadata && fileMetadata[H.ID]) || null;
   }
 
   getDependencies(file: Path): ?Array<string> {
-    const fileMetadata = this._files.get(file);
+    const fileMetadata = this._getFileData(file);
     return (fileMetadata && fileMetadata[H.DEPENDENCIES]) || null;
   }
 
   getSha1(file: Path): ?string {
-    const fileMetadata = this._files.get(file);
+    const fileMetadata = this._getFileData(file);
     return (fileMetadata && fileMetadata[H.SHA1]) || null;
   }
 
   exists(file: Path): boolean {
-    return this._files.has(file);
+    return this._getFileData(file) != null;
   }
 
   getAllFiles(): Array<string> {
-    return Array.from(this._files.keys());
+    return Array.from(this.getFileIterator());
   }
 
-  getFileIterator(): Iterator<string> {
-    return this._files.keys();
+  *getFileIterator(): Iterator<string> {
```

Comment:
woah, fancy.

Reply:
How many times can we use iterators, right? :D